### DISCUSSION TITLE
Input Current from CSV

### DISCUSSION THREAD

**Discussion by kimo707** [pybamm-team/PyBaMM#3015] [Q&A]
Hi,
 I have encountered a problem while attempting to change the input current using the file in drive_cycles WLTC.csv. I have followed the code example provided, but unfortunately, I keep receiving an error message stating, "The corrector convergence failed repeatedly" or "with |h| = hmin." The same issue persists when using other files as well.
Here is the code I have been using:
```python3
model = pybamm.lithium_ion.DFN()
# import drive cycle from file
drive_cycle = pd.read_csv(r"C:\Users\DSR-KI\Downloads\WLTC.csv", delimiter=',', comment="#", header=None).to_numpy()
#drive_cycle[:, 1] = drive_cycle[:, 1]
# load parameter values
param = pybamm.ParameterValues("Chen2020")
current_interpolant = pybamm.Interpolant(drive_cycle[:, 0], drive_cycle[:, 1], pybamm.t)
param["Current function [A]"] = current_interpolant
# set up simulation
solver = pybamm.CasadiSolver(mode="fast")
simulation = pybamm.Simulation(model, parameter_values=param, solver=solver)
# simulate
simulation.solve()
# plot
simulation.plot()
```
I have tried to troubleshoot the problem on my own, but I have been unable to find a solution. Hence, I am reaching out to seek your assistance in resolving this issue. I believe your expertise in this area will greatly help me overcome this hurdle.
Could you kindly provide guidance or suggest any possible solutions that may help fix this error? I would greatly appreciate your input and any instructions you can provide.
Thank you very much for your attention to this matter. I am looking forward to your prompt response.

**Comment by brosaplanella** [comment]
The problem is that the solver doesn't converge. My first guess would be that the drive cycle is too aggressive. The Chen2020 parameter set is for a 5Ah battery which maximum recommended C-rate is 1.5, hence 7.5 A. If the drive cycle is too strong the solver would crash, and I think that's what's happening. Can you check the values of your drive cycle to see if they are too high?

**Reply by brosaplanella** [reply]
This seems related to the PR you opened, so I am closing it.

**Reply by kimo707** [reply]
First thanks for your reply , Yes, you are right. The current value of my cycle is 50 A, which is the reason for your concern. Could you please recommend a solution to this problem? I would like to simulate a drive cycle consisting of five charge and discharge cycles, with the target value reaching 50A/-50A.

**Reply by brosaplanella** [reply]
The way to go is to rescale the current values to give them more reasonable values. By which factor you rescale them is more of a research question (i.e. it depends on what you are trying to model) so unfortunately I can't advise here.
